### PR TITLE
PDA-82 Form Widgets

### PR DESCRIPTION
**Pull request by ffleandro**
Added Support for the following form widgets:
    Combo, Slider, TextArea, Radio, Double, Date, MultipleSelector and InputList;

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: src/main/java/org/entando/plugins/pda/serializer/JsonSchemaFormSerializer.java [hunk 1]
```diff
@@ -10,13 +10,15 @@
...
+import org.entando.plugins.pda.core.model.form.FormFieldDate;
+import org.entando.plugins.pda.core.model.form.FormFieldNumber;
+import org.entando.plugins.pda.core.model.form.FormFieldSelector;
 import org.entando.plugins.pda.core.model.form.FormFieldSubForm;
 import org.entando.plugins.pda.core.model.form.FormFieldText;
 import org.entando.plugins.pda.core.model.form.FormFieldType;
 import org.springframework.stereotype.Component;
 
-@SuppressWarnings("PMD.TooManyMethods")
+@SuppressWarnings({"PMD.TooManyMethods", "PMD.GodClass"})
```
Comment (werbth) [inline src/main/java/org/entando/plugins/pda/serializer/JsonSchemaFormSerializer.java:21]: We should solve GodClass (later)

File: src/main/java/org/entando/plugins/pda/serializer/JsonSchemaFormSerializer.java [hunk 2]
```diff
@@ -155,25 +178,113 @@ private void writeFieldType(JsonGenerator jsonGenerator, String fieldType) throw
         jsonGenerator.writeStringField(TYPE, fieldType);
     }
 
-    private void writeFieldMinMax(JsonGenerator jsonGenerator, FormField field) throws IOException {
-        if (field.getType().equals(FormFieldType.INTEGER)) {
-            FormFieldInteger intField = (FormFieldInteger) field;
-            if (intField.getMinValue() != null) {
-                jsonGenerator.writeNumberField(TYPE_INTEGER_MIN, intField.getMinValue());
+    @SuppressWarnings("PMD.CyclomaticComplexity")
```
Comment (werbth) [inline src/main/java/org/entando/plugins/pda/serializer/JsonSchemaFormSerializer.java:181]: We should solve CyclomaticComplexity (later)

File: src/main/java/org/entando/plugins/pda/serializer/JsonSchemaFormSerializer.java [hunk 3]
```diff
@@ -155,25 +178,113 @@ private void writeFieldType(JsonGenerator jsonGenerator, String fieldType) throw
...
+    }
+
+    private void writeFieldSelector(JsonGenerator jsonGenerator, FormFieldSelector field) throws IOException {
+        if (StringUtils.isNotEmpty(field.getDefaultValue())) {
+            jsonGenerator.writeStringField(DEFAULT, field.getDefaultValue());
+        }
+
+        jsonGenerator.writeFieldName(ONE_OF);
+        jsonGenerator.writeStartArray();
+        for (FormFieldSelector.Option option : field.getOptions()) {
```
Comment (werbth) [inline src/main/java/org/entando/plugins/pda/serializer/JsonSchemaFormSerializer.java]: This for loop is a duplication for the one in the method below, please resolve the duplication.

**Review by werbth — CHANGES_REQUESTED** [review]
@ffleandro Please, resolve only the duplication for now. The other comments we are going to solve later.

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by werbth** [comment]
Retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Review by werbth — APPROVED** [review]

**Pull request merged**